test/sequential/rolldown.withdraw.preOperation.test.ts

```diff
@@ -0,0 +1,578 @@
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    expect(event.assignee).toEqual(testUser.keyRingPair.address);
+    expect(event.source).toEqual("Manual");
+    const nextRequestId = (await Rolldown.getL2RequestsBatchLast()).rangeTo + 1;
+    await Sudo.batchAsSudoFinalized(
+      ...(await Rolldown.createABatchWithWithdrawals(
+        testUser,
+        gaspIdL1Asset.ethereum,
+        batchSize - 1,
+      )),
+    );
+    await signTx(
+      getApi(),
+      await Withdraw(testUser, BN_HUNDRED, gaspIdL1Asset.ethereum, "Ethereum"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const sequencersList = await SequencerStaking.activeSequencers();
+    expect(sequencersList.toHuman().Ethereum).toContain(event.assignee);
+    expect(event.source).toEqual("AutomaticSizeReached");
+    expect(event.range.from.toNumber()).toEqual(nextRequestId);
+    expect(event.range.to.toNumber()).toEqual(nextRequestId + (batchSize - 1));
+  });
+
+  test("GIVEN a manually generated batch, from A,B, WHEN timed happen THEN automatic batch will skip those manual ids", async () => {
+    await Sudo.batchAsSudoFinalized(
+      ...(await Rolldown.createABatchWithWithdrawals(
+        testUser,
+        gaspIdL1Asset.ethereum,
+        batchSize - 1,
+      )),
+    );
+    await signTx(
+      getApi(),
+      await Rolldown.createManualBatch("EthAnvil"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    const eventManually = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    expect(eventManually.assignee).toEqual(testUser.keyRingPair.address);
+    expect(eventManually.source).toEqual("Manual");
+    await Sudo.batchAsSudoFinalized(
+      ...(await Rolldown.createABatchWithWithdrawals(
+        testUser,
+        gaspIdL1Asset.ethereum,
+        batchSize,
+      )),
+    );
+    const eventAutomatically = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const sequencersList = await SequencerStaking.activeSequencers();
+    expect(sequencersList.toHuman().Ethereum).toContain(
+      eventAutomatically.assignee,
+    );
+    expect(eventAutomatically.source).toEqual("AutomaticSizeReached");
+    expect(eventAutomatically.range.from.toNumber()).toBeGreaterThan(
+      eventManually.range.to.toNumber(),
+    );
```

Comment:
this test is about validating that the automatic .From and .To do not include the previous manually generated batches. plz assert that.

Reply:
This is what I am checking here. We are checking that the automatic packaging follows the manual one, that is why its range numbers follow the manual ones.

Reply:
you are right, It is quite tricky to guess from here `txBatchEvent[0].event.data[4][1],`  but I think u tested carefully that this event number is the .TO in the previous batch.